From 88f10aed35c19cc43a88bbcd12b026cb32c0003f Mon Sep 17 00:00:00 2001
From: Eike Rathke <erack@redhat.com>
Date: Sun, 7 Oct 2018 00:24:31 +0200
Subject: Resolves: tdf#73081 empty <> not-empty is also a match if numeric
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This simplifies to match empty for the same conditions whether
numeric or not, an empty string does not result in numeric 0 here.

Change-Id: I923a3e1f07dd2691d3ac720fc28ba76986fe6c3d
Reviewed-on: https://gerrit.libreoffice.org/61486
Reviewed-by: Eike Rathke <erack@redhat.com>
Tested-by: Jenkins
(cherry picked from commit 9d8c89d9044edb5ad5c8bc4a8f0fd5868192714e)
Reviewed-on: https://gerrit.libreoffice.org/61495
Reviewed-by: Caolán McNamara <caolanm@redhat.com>
Tested-by: Caolán McNamara <caolanm@redhat.com>
---
 sc/source/core/tool/queryparam.cxx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sc/source/core/tool/queryparam.cxx b/sc/source/core/tool/queryparam.cxx
index 269eb70..6e2fefa 100644
--- a/sc/source/core/tool/queryparam.cxx
+++ b/sc/source/core/tool/queryparam.cxx
@@ -290,9 +290,9 @@ void ScQueryParamBase::FillInExcelSyntax(
          * This could be handled independently if all queries should support
          * it, needs to be evaluated if that actually is desired. */
 
-        // (empty = empty) is a match, and (empty <> not-empty) also is a match
-        if (rItem.meType == ScQueryEntry::ByString)
-            rItem.mbMatchEmpty = ((rEntry.eOp == SC_EQUAL && rItem.maString.isEmpty())
+        // (empty = empty) is a match, and (empty <> not-empty) also is a
+        // match. (empty = 0) is not a match.
+        rItem.mbMatchEmpty = ((rEntry.eOp == SC_EQUAL && rItem.maString.isEmpty())
                 || (rEntry.eOp == SC_NOT_EQUAL && !rItem.maString.isEmpty()));
     }
 }
-- 
cgit v1.1
